conans/client/build/cmake_flags.py

```diff
@@ -294,9 +295,12 @@ def get_definitions(self):
         definitions.update(build_type_definition(self._forced_build_type, build_type,
                                                  self._generator, self._output))
 
-        if str(os_) == "Macos":
-            if arch == "x86":
-                definitions["CMAKE_OSX_ARCHITECTURES"] = "i386"
+        if tools.is_apple_os(os_):
+            definitions["CMAKE_OSX_ARCHITECTURES"] = tools.to_apple_arch(arch)
+            # xcrun is only available on macOS, otherwise it's cross-compiling and it needs to be
+            # set within CMake toolchain
+            if platform.system == "Darwin":
```

Comment:
Isn't this missing a ()?

``platform.system() == "Darwin"``

Maybe the test patch is hiding this.